src/zenml/integrations/huggingface/deployers/huggingface_deployer.py

```diff
@@ -0,0 +1,545 @@
+
+    def _get_entrypoint_and_command(
+        self, deployment: DeploymentResponse
+    ) -> Tuple[str, str]:
+        """Generate ENTRYPOINT and CMD for the Dockerfile.
+
+        Args:
+            deployment: The deployment.
+
+        Returns:
+            Tuple of (ENTRYPOINT line, CMD line) for Dockerfile.
+        """
+        # Get entrypoint command: ["python", "-m", "zenml.entrypoints.entrypoint"]
+        entrypoint = DeploymentEntrypointConfiguration.get_entrypoint_command()
+
+        # Get arguments with deployment ID
+        arguments = DeploymentEntrypointConfiguration.get_entrypoint_arguments(
+            **{DEPLOYMENT_ID_OPTION: deployment.id}
+        )
+
+        # Format as JSON arrays for Dockerfile exec form
+        # Use json.dumps() to ensure proper JSON with double quotes
+        entrypoint_line = f"ENTRYPOINT {json.dumps(entrypoint)}"
+        cmd_line = f"CMD {json.dumps(arguments)}"
+
+        return entrypoint_line, cmd_line
+
+    def _generate_image_reference_dockerfile(
+        self,
+        image: str,
+        deployment: DeploymentResponse,
+    ) -> str:
+        """Generate Dockerfile that references a pre-built image.
+
+        Note: Environment variables and secrets are NOT included in the
+        Dockerfile for security reasons. They are set using Hugging Face's
+        Space secrets and variables API instead.
+
+        Args:
+            image: The pre-built image to reference.
+            deployment: The deployment.
+
+        Returns:
+            The Dockerfile content.
+        """
+        lines = [f"FROM {image}"]
+
+        # Add user
+        lines.append("USER 1000")
+
+        # Add entrypoint and command
+        entrypoint_line, cmd_line = self._get_entrypoint_and_command(
+            deployment
+        )
+        lines.append(entrypoint_line)
+        lines.append(cmd_line)
+
+        return "\n".join(lines)
+
+    def do_provision_deployment(
+        self,
+        deployment: DeploymentResponse,
+        stack: "Stack",
+        environment: Dict[str, str],
+        secrets: Dict[str, str],
+        timeout: int,
+    ) -> DeploymentOperationalState:
+        """Provision a Huggingface Space deployment.
+
+        Args:
+            deployment: The deployment to run.
+            stack: The active stack.
+            environment: Environment variables for the app.
+            secrets: Secret environment variables for the app.
+            timeout: Maximum time to wait for deployment (unused).
+
+        Returns:
+            Operational state of the provisioned deployment.
+
+        Raises:
+            DeploymentProvisionError: If the deployment cannot be provisioned.
+        """
+        assert deployment.snapshot, "Pipeline snapshot not found"
+
+        settings = cast(
+            HuggingFaceDeployerSettings,
+            self.get_settings(deployment.snapshot),
+        )
+
+        api = self._get_hf_api()
+        space_id = self._get_space_id(deployment)
```

Comment:
There's a corner-case here that you haven't addressed: if there's an existing space ID stored in `deployment.deployment_metadata["space_id"]` that doesn't match the new space ID, for example because the deployment was renamed or the prefix settings changed. When this happens, you could simply call `do_deprovision_deployment` to clean up the previous space.